src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dns/ClusterDnsCache.scala

```diff
@@ -51,41 +43,29 @@ object ClusterDnsCache {
   * @param proxyConfig the proxy configuration
   * @param dbRef provides access to the database
   */
-class ClusterDnsCache(proxyConfig: ProxyConfig, dbRef: DbReference) extends Actor with LazyLogging {
-  var ProjectNameToHost: Map[(GoogleProject, ClusterName), GetClusterResponse] = Map.empty
-
-  import context.dispatcher
-
-  override def preStart(): Unit = {
-    super.preStart()
-    self ! RefreshFromDatabase
-  }
-
-  override def receive: Receive = {
-    case RefreshFromDatabase =>
-      queryForClusters pipeTo self
-
-    case ProcessClusters(clusters) =>
-      processClusters(clusters)
-      scheduleRefresh
-
-    // add one ready cluster to the caches now instead of waiting for the refresh cycle
-    case ProcessReadyCluster(cluster) =>
-      sender ! processReadyCluster(cluster)
-
-    case GetByProjectAndName(googleProject, clusterName) =>
-      sender ! ProjectNameToHost.getOrElse((googleProject, clusterName), ClusterNotFound)
-  }
-
-  private def scheduleRefresh = {
-    context.system.scheduler.scheduleOnce(proxyConfig.dnsPollPeriod, self, RefreshFromDatabase)
-  }
-
-  private def queryForClusters: Future[ProcessClusters] ={
-    dbRef.inTransaction { dataAccess =>
-      dataAccess.clusterQuery.listActive()
-    }.map(ProcessClusters.apply)
-  }
+class ClusterDnsCache(proxyConfig: ProxyConfig, dbRef: DbReference, dnsCacheConfig: ClusterDnsCacheConfig)(implicit executionContext: ExecutionContext) extends LazyLogging {
+
+
+
+  val projectNameToHost = CacheBuilder.newBuilder()
+    .expireAfterWrite(dnsCacheConfig.cacheExpiryTime.toSeconds, TimeUnit.SECONDS)
+    .maximumSize(dnsCacheConfig.cacheMaxSize)
+    .build(
+      new CacheLoader[DnsCacheKey, Future[HostStatus]] {
+        def load(key: DnsCacheKey) = {
+          dbRef.inTransaction {dataAccess => dataAccess.clusterQuery.getActiveClusterByName(key.googleProject, key.clusterName)}.map {
+              case  Some(cluster) => {
+                val res = projectNameToHostEntry(cluster)._2
+                if(res.isInstanceOf[HostReady]) {
+                  ClusterDnsCache.HostToIp = Map(hostToIpEntry(cluster))
```

Comment:
Already talked to Vikram about this but for a reminder for myself...we're gonna want to add the new entry rather than overwriting the Map content.

Reply:
I also wonder if we can write a unit test that would fail with the code line above and pass with the change?